packages/drivers/odsp-driver/src/odspDriverUrlResolver2.ts

```diff
@@ -84,14 +85,14 @@ export class OdspDriverUrlResolver2 implements IUrlResolver {
 
         const odspResolvedUrl = await resolveRequest(requestToBeResolved);
 
-        // Generate sharingLink only if specified in the request.
-        if (requestToBeResolved.headers?.[SharingLinkHeader.generateSharingLink]) {
-            await this.getShareLinkPromise(odspResolvedUrl)
-            .then((shareLink: string) => sharingLink = shareLink)
-            .catch(() => {});
+        // Generate sharingLink only if main url is not sharing link.
+        if (isSharingLink !== true) {
```

Comment:
this should be equivalent to if(isSharingLink) :)

Reply:
my bad